src/llama_stack/core/server/server.py

```diff
@@ -526,8 +526,8 @@ def extract_path_params(route: str) -> list[str]:
 
 def remove_disabled_providers(obj):
     if isinstance(obj, dict):
-        keys = ["provider_id", "shield_id", "provider_model_id", "model_id"]
-        if any(k in obj and obj[k] in ("__disabled__", "", None) for k in keys):
+        # Filter out items where provider_id is explicitly disabled or empty
```

Comment:
yes we had added this behavior when our starter run.yaml had models also listed and disabled via these methods. I _believe_ this is now not needed but if you could double check environment variables which end up enabling or disabling certain entries in run.yaml which would cause this block to trigger, that would be great.

Reply:
Tested with the starter distro, PR updated slightly (added None to the list), extract_path_params now correctly filters out providers disabled by the absence of a env variable.